include/dynd/types/string_type.hpp

```diff
@@ -22,6 +22,8 @@ class DYND_API string : public bytes {
 
   string(const std::string &other) : string(other.data(), other.size()) {}
 
+  string(const bytes &other) : string(other.data(), other.size()) {}
+
   bool operator<(const string &rhs) const
```

Comment:
Why is `string(const bytes &other)` necessary? It feels strange to me to have a copy constructor in string from it's base class.